eessi/testsuite/tests/libs/blas/blas.py

```diff
@@ -0,0 +1,272 @@
+"""
+This test is adapted from the BLAS test included in the BLIS v1.1 sources at
+https://github.com/flame/blis/tree/1.1/test/3
+
+Customizations to the original BLAS test:
+- adapted and simplified Makefile for FlexiBLAS support
+- custom simplified run.sh script
+
+Supported tags in this ReFrame test (in addition to the common tags):
+- threading: `st`, `mt`
+- BLAS implementation: `openblas`, `blis`, `aocl-blas`, `mkl`
+- toolchain: `2023a`, `2023b`, `2024a`, ...
+- `CI` tag: runs only openblas st + mt
+"""
+
+
+import reframe as rfm
+from reframe.core.backends import getlauncher
+from reframe.core.builtins import parameter, run_after, run_before, sanity_function
+import reframe.utility.sanity as sn
+
+from eessi.testsuite.constants import COMPUTE_UNITS, DEVICE_TYPES, SCALES
+from eessi.testsuite.eessi_mixin import EESSI_Mixin
+from eessi.testsuite.utils import check_modules_avail, log
+
+BLAS_MODULES = {
+    '2023a': {
+        'FlexiBLAS': 'FlexiBLAS/3.3.1-GCC-12.3.0',
```

Comment:
What we could do is use the `find_modules` to search for `FlexiBLAS`, and then based on the toolchain, for each FlexiBLAS that is found, we can find the compatible BLIS modules (i.e. the one that is built with the same toolchain). Based on that, you can then create a parameter space of FlexiBLAS + BLIS modules that should be loaded.

Reply:
done in https://github.com/EESSI/test-suite/pull/268/commits/cb446196d5b5dbcb0d073da3d45c12772d8b9c38